core/src/test/scala/org/apache/spark/launcher/LauncherBackendSuite.scala

```diff
@@ -46,15 +46,22 @@ class LauncherBackendSuite extends SparkFunSuite with Matchers {
   private def testWithMaster(master: String): Unit = {
     val env = new java.util.HashMap[String, String]()
     env.put("SPARK_PRINT_LAUNCH_COMMAND", "1")
-    val handle = new SparkLauncher(env)
+    val launcher = new SparkLauncher(env)
       .setSparkHome(sys.props("spark.test.home"))
       .setConf(SparkLauncher.DRIVER_EXTRA_CLASSPATH, System.getProperty("java.class.path"))
       .setConf(UI_ENABLED.key, "false")
       .setConf(SparkLauncher.DRIVER_EXTRA_JAVA_OPTIONS, s"-Dtest.appender=console")
       .setMaster(master)
       .setAppResource(SparkLauncher.NO_RESOURCE)
       .setMainClass(TestApp.getClass.getName().stripSuffix("$"))
-      .startApplication()
+
+    if(JavaModuleUtils.isJavaVersionAtLeast17) {
```

Comment:
Rather than just add these in tests, can we ensure they get added for Java 17 in the non-test code paths? like ensure that these args are always added to the JVM command? that would go a long way to not disrupting users

Reply:
I agree with @LuciferYang (https://github.com/apache/spark/pull/34153#discussion_r720688637 ).
@srowen , why don't we add these options `-XX:+IgnoreUnrecognizedVMOptions --add-opens=java.base/java.nio=ALL-UNNAMED ...` always for all JDKs at Apache Spark 3.3? We can remove them later.

Reply:
Agree. We can set it wherever possible.

Reply:
And nit: space after if

Reply:
Thank you, @srowen .